src/quick-lint-js/parse.h

```diff
@@ -408,13 +414,15 @@ class parser {
   template <QLJS_PARSE_VISITOR Visitor>
   void visit_compound_assignment_expression(expression_ptr lhs,
                                             expression_ptr rhs, Visitor &v) {
-    this->visit_expression(lhs, v, variable_context::rhs);
+    bool used = false;
+    used = this->visit_expression(lhs, v, variable_context::rhs, true);
```

Comment:
Instead of having `visit_expression` take a `bool` and return a `bool`, what if we do this:
```suggestion
    bool used = false;
    if (lhs->kind() == expression_kind::variable) {
      used = true;
    } else {
      this->visit_expression(lhs, v, variable_context::rhs);
    }
```

Reply:
Done.